server/controllers/article.controller.js

```diff
@@ -20,25 +20,123 @@ class ArticleController {
+      title, body, tagsList, description, published
     } = req.body;
     const { id } = req.user;
     const successMessage = 'Article successfully created';
     const errorMessage = 'Article was not created successfully';
+    const tags = [...new Set(tagsList)];
     const article = await Article.create({
       title,
       body,
-      tags,
+      tagsList,
       description,
-      image,
+      image: req.body.image,
       published,
-      slug: slug(`${title} ${Date.now()}`),
+      slug: slugify(`${title} ${Date.now()}`),
       userId: id
     });
+    if (tags) {
+      let insertTags = tags.map(async (newTags) => {
+        const newtag = await tag.findOrCreate({ where: { name: newTags } });
+        return newtag[0];
+      });
+
+      insertTags = await Promise.all(insertTags);
+
+      await article.addTags(insertTags);
+    }
+    const {
+      dataValues: {
+        firstname,
+        email,
+        bio,
+        image
+      }
+    } = await article.getUser();
+    const author = {
+      firstname,
+      email,
+      bio,
+      image
+    };
     if (!article) {
       return Response(res, 400, errorMessage);
     }
-    return Response(res, 201, successMessage, { article });
+    return Response(res, 201, successMessage, { article, author });
+  }
+
+  /**
+   * Edit article controller
+   * @param {Object} req
+   * @param {Object} res
+   * @returns {Object} response
+   */
+  static async editArticle(req, res) {
+    const {
+      params: { slug },
+      body: {
+        description, title, body, tagsList
+      }, user: { id }
+    } = req;
+    const tags = [...new Set(tagsList)];
+    const editarticle = await Article.update({
+      description,
+      title,
+      body,
+      tagsList
+    }, {
+      where: {
+        slug, userId: id,
+      },
+      returning: true,
+    });
+    let insertTags;
+    const articleId = editarticle[1][0].dataValues.id;
+    const editInstance = editarticle[1][0];
+    if (tags) {
+      await articletag.destroy({ where: { articleId } });
+      insertTags = tags.map(async (newTags) => {
+        const newtag = await tag.findOrCreate({ where: { name: newTags } });
+        return newtag[0];
+      });
+
+      insertTags = await Promise.all(insertTags);
+      await editInstance.addTags(insertTags);
+    }
+    const {
+      dataValues: {
+        firstname,
+        email,
+        bio,
+        image
+      }
+    } = await editInstance.getUser();
+    const {
+      dataValues: {
+        updatedAt
+      }
+    } = editarticle[1][0];
+    const response = {
+      article: {
+        slug,
+        title,
+        description,
+        body,
+        tagsList,
+        updatedAt,
+      },
+      author: {
+        firstname,
+        email,
+        bio,
+        image
+      }
+    };
+    if (!editarticle) {
+      return Response(res, 400, 'error editing article');
+    }
+    return Response(res, 200, 'edited successfully', response);
```

Comment:
What was edited successfully? @davidokonji

Reply:
Would make the response more descriptive